### PR TITLE
Add notes about endgame for test flakes

### PR DESCRIPTION
**Pull request by pmorie**
@lavalamp as we discussed last week, sorry for the latency.

### CODE REVIEW

**Comment by bgrant0607** [comment]
@pmorie You can assign PRs when you create them. :-)

**Comment by pmorie** [comment]
@bgrant0607 forgive me, I am distracted, preparing for a showing on my house :)

File: docs/devel/flaky-tests.md [hunk 1]
```diff
@@ -92,7 +92,14 @@ we have the following guidelines:
...
    progress towards narrowing it down every time a flake occurs. If you can't
    figure it out from the logs, add log messages that would have help you figure
-   it out.
+   it out.  If you make changes to make a flake more reproducible, please link
+   your pull request to the flake you're working on.
+5. Once you have made a change that you believe fixes a flake, it is conservative
+   to keep the issue for the flake open and see if it manifests again after the
+   change is merged.
+6. If a flake has been open, could not be reproduced, and has not manifested in
+   2 months or so, it is reasonable to close the flake issue with a note saying
```
Comment (lavalamp) [inline docs/devel/flaky-tests.md]: s/2 months or so/3 months/

File: docs/devel/flaky-tests.md [hunk 2]
```diff
@@ -92,7 +92,14 @@ we have the following guidelines:
...
+   why.
```
Comment (lavalamp) [inline docs/devel/flaky-tests.md]: 7. PRs that fix or help debug flakes may have the P0 priority set to get them through the merge queue as fast as possible.

**Comment by lavalamp** [comment]
Could go in as is, but since I had a nit, I asked for a pony, too.

**Comment by pmorie** [comment]
@lavalamp :carousel_horse:

**Comment by pmorie** [comment]
updated, PTAL

**Comment by lavalamp** [comment]
LGTM, thanks!

**Pull request merged**